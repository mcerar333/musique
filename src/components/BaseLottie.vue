<template>
  <div class="lottie-container" ref="lottie"></div>
</template>

<script>
import lottie from 'lottie-web';

export default {
  name: 'BaseLottie',

  props: {
    options: {
      type: Object,
      required: true,
    },
  },

  data() {
    return {
      anim: null,
    };
  },
  mounted() {
    this.anim = lottie.loadAnimation({
      renderer: 'svg',
      container: this.$refs.lottie,
      loop: this.options.loop ?? true,
      autoplay: this.options.autoplay ?? true,
      animationData: this.options.animationData,
      rendererSettings: this.options.rendererSettings,
    });
  },
  beforeUnmount() {
    lottie.destroy();
  },
};
</script>

<style lang="scss">
.lottie-container {
  max-width: 100%;
  max-height: 100%;
  overflow: hidden;
}
</style>
