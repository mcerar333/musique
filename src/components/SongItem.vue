<template>
  <li class="song">
    <div class="song__info">
      <router-link :to="{ name: 'Song', params: { id: song.docID } }">{{
        songName
      }}</router-link>
      <p>{{ song.displayName }}</p>
    </div>

    <router-link
      :to="{ name: 'Song', params: { id: song.docID }, hash: '#comments' }"
      v-slot="{ navigate }"
      custom
    >
      <div class="song__comments" v-svg="'icon-chat'" @click="navigate">
        <span>{{ song.commentCount }}</span>
      </div>
    </router-link>
  </li>
</template>

<script>
export default {
  name: 'SongItem',

  props: {
    song: {
      type: Object,
      required: true,
    },
  },
  computed: {
    songName() {
      return this.song.modifiedName.replace('.mp3', '');
    },
  },
};
</script>
