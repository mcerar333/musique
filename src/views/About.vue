<template>
  <section class="about">
    <div class="about__info">
      <h1 class="about__title">{{ $t('about.title') }}</h1>

      <p>{{ $t('about.p1') }}</p>
      <p>{{ $t('about.p2') }}</p>
      <p>{{ $t('about.p3') }}</p>
    </div>

    <figure class="about__image">
      <img src="/assets/img/girl.png" alt="girl listening to music" />
    </figure>
  </section>
</template>

<script>
export default {
  name: 'About',
};
</script>
