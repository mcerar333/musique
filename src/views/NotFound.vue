<template>
  <section class="not-found">
    <BaseLottie :options="lottieOptions" />
  </section>
</template>

<script>
import BaseLottie from '@/components/BaseLottie.vue';
import * as animationData from '@/assets/lottie/error.json';

export default {
  name: 'NotFound',
  components: { BaseLottie },

  data() {
    return {
      lottieOptions: { animationData },
    };
  },
};
</script>
