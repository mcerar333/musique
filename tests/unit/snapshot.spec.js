import { shallowMount, RouterLinkStub } from '@vue/test-utils';

import SongItem from '@/components/SongItem.vue';
import Svg from '@/directives/svg';

import NotFound from '@/views/NotFound.vue';
import BaseLottie from '@/components/BaseLottie.vue';
import * as animationData from '@/assets/lottie/error.json';

import About from '@/views/About.vue';

describe('Snapshots SongItem.vue', () => {
  test('component renders correctly', () => {
    const song = {
      docID: '123abc',
      displayName: 'Bobby Brown',
      modifiedName: 'Stairway to Heaven',
      commentCount: 15,
    };

    const wrapper = shallowMount(SongItem, {
      props: {
        song,
      },
      global: {
        components: {
          'router-link': RouterLinkStub,
        },
        directives: {
          Svg,
        },
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});

jest.mock('lottie-web', () => ({
  loadAnimation: () => true,
}));

describe('Snapshots BaseLottie.vue', () => {
  test('component renders correctly', () => {
    const wrapper = shallowMount(BaseLottie, {
      props: {
        options: { animationData },
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});

describe('Snapshots NotFound.vue', () => {
  test('component renders correctly', () => {
    const wrapper = shallowMount(NotFound, {
      components: { BaseLottie },

      data() {
        return {
          lottieOptions: { animationData },
        };
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});

describe('Snapshots About.vue', () => {
  test('component renders correctly', () => {
    const wrapper = shallowMount(About, {
      global: {
        mocks: {
          $t: msg => msg,
        },
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
