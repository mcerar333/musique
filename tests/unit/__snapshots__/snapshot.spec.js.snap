// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots About.vue component renders correctly 1`] = `
<section
  class="about"
>
  <div
    class="about__info"
  >
    <h1
      class="about__title"
    >
      about.title
    </h1>
    <p>
      about.p1
    </p>
    <p>
      about.p2
    </p>
    <p>
      about.p3
    </p>
  </div>
  <figure
    class="about__image"
  >
    <img
      alt="girl listening to music"
      src="/assets/img/girl.png"
    />
  </figure>
</section>
`;

exports[`Snapshots BaseLottie.vue component renders correctly 1`] = `
<div
  class="lottie-container"
/>
`;

exports[`Snapshots NotFound.vue component renders correctly 1`] = `
<section
  class="not-found"
>
  <base-lottie-stub
    options="[object Object]"
  />
</section>
`;

exports[`Snapshots SongItem.vue component renders correctly 1`] = `
<li
  class="song"
>
  <div
    class="song__info"
  >
    <router-link-stub
      to="[object Object]"
    />
    <p>
      Bobby Brown
    </p>
  </div>
  <router-link-stub
    custom=""
    to="[object Object]"
  />
</li>
`;
